### PR TITLE
MGDOBR-721: ManagedResource: Validate and trim name

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-721](https://issues.redhat.com/browse/MGDOBR-721)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/models/requests/BridgeRequest.java [hunk 1]
```diff
@@ -32,11 +34,11 @@ public BridgeRequest(String name, Action errorHandler) {
     }
 
     public Bridge toEntity() {
-        return new Bridge(name);
+        return new Bridge(getName());
     }
 
     public String getName() {
-        return name;
+        return Objects.nonNull(name) ? name.trim() : null;
```
Comment (manstis) [inline manager/src/main/java/com/redhat/service/smartevents/manager/api/models/requests/BridgeRequest.java:41]: I elected to trim on the getter. This class has no setter.

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/models/requests/ProcessorRequest.java [hunk 2]
```diff
@@ -69,45 +70,25 @@ public ProcessorType getType() {
     }
 
     public String getName() {
-        return name;
-    }
-
-    public void setName(String name) {
-        this.name = name;
+        return Objects.nonNull(name) ? name.trim() : null;
```
Comment (manstis) [inline manager/src/main/java/com/redhat/service/smartevents/manager/api/models/requests/ProcessorRequest.java:73]: I elected to trim on the getter. I have removed the setters.

**Comment by manstis** [comment]
@sutaakar Please feel to reassign.

**Review by r00ta — APPROVED** [review]

**Review by jomarko — COMMENTED** [review]
@manstis thank you for the PR. I tried to create new bridge with name "  xxx   " and it was auto trimmed to "xxx". I have just a  question, should user find some message about it calling errors api? i.e. "http://localhost:8080/api/v1/errors" ?

Something like:
```
    {
      "kind": "Error",
      "id": "2",
      "href": "/api/v1/errors/2",
      "code": "OPENBRIDGE-2",
      "reason": "SmartEvent instance name was not valid"
    },
```

**Comment by manstis** [comment]
Hi @jomarko 

> ...should user find some message about it calling errors api? i.e. "http://localhost:8080/api/v1/errors" ?

Sorry, I do not understand. Are you suggesting an error should be returned?

**Review by jomarko — APPROVED** [review]
Then no other action can be taken , than approval. Thank you @manstis

**Comment by manstis** [comment]
Two approvals obtained; merging.

**Pull request merged**